substrate/frame/distribution/src/functions.rs

```diff
@@ -0,0 +1,110 @@
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Helper functions for Distribution pallet.
+
+pub use super::*;
+impl<T: Config> Pallet<T> {
+	pub fn pot_account() -> AccountIdOf<T> {
+		// Get Pot account
+		let pot_id = T::PotId::get();
+		let pot_account: AccountIdOf<T> = pot_id.into_account_truncating();
+		pot_account
+	}
+
+	/// Series of checks on the Pot, to ensure that we have enough funds
+	/// before executing a Spend
+	pub fn pot_check(spend: BalanceOf<T>) -> DispatchResult {
+		// Get Pot account
+		let pot_account: AccountIdOf<T> = Self::pot_account();
+
+		// Check that the Pot as enough funds for the transfer
+		let balance = T::NativeBalance::balance(&pot_account);
+		let minimum_balance = T::NativeBalance::minimum_balance();
+		let remaining_balance = balance.saturating_sub(spend);
+
+		ensure!(remaining_balance > minimum_balance, Error::<T>::InsufficientPotReserves);
+		ensure!(balance > spend, Error::<T>::InsufficientPotReserves);
+		Ok(())
+	}
+
+	/// Funds transfer from the Pot to a project account
+	pub fn spend(amount: BalanceOf<T>, beneficiary: AccountIdOf<T>) -> DispatchResult {
+		// Get Pot account
+		let pot_account: AccountIdOf<T> = Self::pot_account();
+
+		//Operate the transfer
+		T::NativeBalance::transfer(&pot_account, &beneficiary, amount, Preservation::Preserve)
+			.map_err(|_| Error::<T>::TransferFailed)?;
+
+		Ok(())
+	}
+
+	// Done in begin_block
+	// At the beginning of every Epoch, populate the `Spends` storage from the `Projects` storage
+	// (populated by an external process/pallet) make sure that there is enough funds before
+	// creating a new `SpendInfo`, and `ProjectInfo` corresponding to a created `SpendInfo`
+	// should be removed from the `Projects` storage. This is also a good place to Reserve the
+	// funds for created `SpendInfos`. the function will be use in a hook.
+
+	pub fn begin_block(now: BlockNumberFor<T>) -> Weight {
+		let max_block_weight = Weight::from_parts(1000_u64, 0);
+		let epoch = T::EpochDurationBlocks::get();
+
+		//We reach the check period
+		if (now % epoch).is_zero() {
+			let mut projects = Projects::<T>::get();
+
+			if projects.len() > 0 {
+				for project in projects.clone() {
+					// check if the pot has enough fund for the Spend
+					let check = Self::pot_check(project.amount);
+					if check.is_ok() {
+						// Create a new Spend
+						let new_spend = SpendInfo::<T>::new(&project);
+
+						// Reserve funds for the project
+						let pot = Self::pot_account();
+						let _ = T::NativeBalance::hold(
+							&HoldReason::FundsReserved.into(),
+							&pot,
+							project.amount,
+						)
+						.expect("Funds Reserve Failed");
```

Comment:
this is invalid usage of expect. it should be a proof stating why it can never fail. in this case, it could be "pot_check already ensured enough funds; qed". However, it is still a bad code as people have to read the inner code of `pod_check` to check if this is true or not.